fili-core/src/main/java/com/yahoo/bard/webservice/web/FiliResponseWriterSelector.java

```diff
@@ -38,16 +39,12 @@ public FiliResponseWriterSelector(
      * @return Response writer for the given format type
      */
     @Override
-    public ResponseWriter select(ApiRequest request) {
+    public Optional<ResponseWriter> select(ApiRequest request) {
         ResponseFormatType format = request.getFormat();
         if (format == null) {
             format = ResponseFormatType.JSON;
         }
-        return writers.get(format);
-    }
-
-    public Map<ResponseFormatType, ResponseWriter> getWriters() {
-        return writers;
+        return Optional.of(writers.get(format));
```

Comment:
This should be `Optional.ofNullable`. `Optional.of` explodes if the value it attempts to write is `null` (which will happen if the format the user provided isn't in the map). `Optional.ofNullable` meanwhile will return an `Optional.empty` if the wrapped value is `null`.